src/main/java/com/blamejared/crafttweaker/impl/predicate/GameMode.java

```diff
@@ -0,0 +1,61 @@
+package com.blamejared.crafttweaker.impl.predicate;
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraft.world.GameType;
+import org.openzen.zencode.java.ZenCodeType;
+
+/**
+ * Represents the various kinds of game modes that are available within the game.
+ *
+ * A game mode indicates how the player should experience the game and with which restrictions. Refer to the
+ * documentation of the various entries for a description.
+ */
+@ZenRegister
+@ZenCodeType.Name("crafttweaker.api.predicate.GameMode")
+@Document("vanilla/api/predicate/GameMode")
+public enum GameMode {
```

Comment:
What about `GameType.NOT_SET`?

I'm not sure how it is actually used in vanilla, but I know it does exist

Reply:
It's just used as a default value instead of `null` in some places. The only "meaningful" use is to store the previous game mode of a player, all the other ones are filtering the game mode out or throwing exceptions in case it matches defaults. In the context of `Predicate`s, it act as "don't care".